pkgs/tools/admin/pgadmin/default.nix

```diff
@@ -0,0 +1,167 @@
+{ stdenv
+, lib
+, python3
+, fetchurl
+, zlib
+, mkYarnModules
+, sphinx
+, nixosTests
+}:
+
+let
+
+  pname = "pgadmin";
+  version = "6.3";
+
+  src = fetchurl {
+    url = "https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v${version}/source/pgadmin4-${version}.tar.gz";
+    sha256 = "0xcg5gx6hf4k15idjkxzsnm5s9829fn4li54ga9qbyfqn3wwpg0h";
+  };
+
+  yarnDeps = mkYarnModules {
+    pname = "${pname}-yarn-deps";
+    inherit version;
+    packageJSON = ./package.json;
+    yarnLock = ./yarn.lock;
+    yarnNix = ./yarn.nix;
+  };
+in
+
+python3.pkgs.buildPythonApplication rec {
+  inherit pname version src;
+
+  # from Dockerfile
+  CPPFLAGS = "-DPNG_ARM_NEON_OPT=0";
+
+  format = "other";
+
+  postPatch = ''
+    # patching Makefile, so it doesn't try to build sphinx documentation here
+    # (will do so later)
+    substituteInPlace Makefile --replace "LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8 $(MAKE) -C docs/en_US -f Makefile.sphinx html" "true"
+    # fix document which refers a non-existing document and fails
+    substituteInPlace docs/en_US/contributions.rst --replace "code_snippets" ""
+    patchShebangs .
+    # relax dependencies
+    substituteInPlace requirements.txt \
+      --replace "Pillow==8.3.*" "Pillow>=8.3.0" \
+      --replace "psycopg2==2.8.*" "psycopg2>=2.8.0" \
+      --replace "cryptography==3.*" "cryptography>=3.0" \
+      --replace "requests==2.25.*" "requests>=2.25.0"
+    # don't use Server Mode (can be overridden later)
+    substituteInPlace pkg/pip/setup_pip.py \
+      --replace "req = req.replace('psycopg2', 'psycopg2-binary')" "req = req" \
+      --replace "builtins.SERVER_MODE = None" "builtins.SERVER_MODE = False"
+  '';
+
+  buildPhase = ''
+    # Adapted from pkg/pip/build.sh
+    echo Creating required directories...
+    mkdir -p pip-build/pgadmin4/docs
+
+    # build the documentation
+    cd docs/en_US
+    ${sphinx}/bin/sphinx-build -W -b html -d _build/doctrees . _build/html
+
+    # Build the clean tree
+    cd ../../web
+    cp -r * ../pip-build/pgadmin4
+    cd ../docs
+    cp -r * ../pip-build/pgadmin4/docs
+    for DIR in `ls -d ??_??/`
+    do
+      if [ -d ''${DIR}_build/html ]; then
+          mkdir -p ../pip-build/pgadmin4/docs/''${DIR}_build
+          cp -Rv ''${DIR}_build/html ../pip-build/pgadmin4/docs/''${DIR}_build
+      fi
+    done
+    cd ../
+
+    cp -r ${yarnDeps}/* pip-build/pgadmin4
+
+    echo Creating distro config...
+    echo HELP_PATH = \'../../docs/en_US/_build/html/\' > pip-build/pgadmin4/config_distro.py
+    echo MINIFY_HTML = False >> pip-build/pgadmin4/config_distro.py
+
+    echo Creating manifest...
+    echo recursive-include pgadmin4 \* > pip-build/MANIFEST.in
+
+    echo Building wheel...
+    cd pip-build
+    ${python3}/bin/python ../pkg/pip/setup_pip.py bdist_wheel
```

Comment:
the default builder does this. The above could be put in `preBuild`

Reply:
thanks for your input. I switched to the default builder (setuptools)